addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/XSLTInjection.java

```diff
@@ -0,0 +1,260 @@
+        "compilation error", "XSLT compile error", "SAXParseException"
+    };
+
+    private static final String[] vendorReturningPayloads = {
+        "<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/>",
+        "system-property(\'xsl:vendor\')/>",
+        "\"/><xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--",
+        "<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--"
+    };
+
+    private static final String[] evidenceVendors = {
+        "libxslt", "Microsoft", "Saxonica", "Apache", "Xalan", "SAXON", "Transformiix"
+    };
+
+    private static final String[] evidencePortScanning = {
+        "failed to open stream",
+        "Invalid Http response",
+        "Connection Refused",
+        "No connection could be made because the target machine actively refused it",
+        "Can not load requested doc"
+    };
+
+    private static final String[] cmdExecPayloads = {
+        "<xsl:variable name=\"rtobject\" select=\"runtime:getRuntime()\"/>\n"
+                + "<xsl:variable name=\"process\" select=\"runtime:exec($rtobject,'erroneous_command')\"/>\n"
+                + "<xsl:variable name=\"waiting\" select=\"process:waitFor($process)\"/>\n"
+                + "<xsl:value-of select=\"$process\"/>",
+        "<xsl:value-of select=\"php:function('exec','erroneous_command 2>&amp;1')\"/>"
+    };
+
+    private static final String[] evidenceCmdExec = {
+        "Cannot run program", "erroneous_command: not found",
+    };
+
+    private enum XSLTInjectionType {
+        ERROR(errorCausingPayloads, evidenceError, "error"),
+        VENDOR(vendorReturningPayloads, evidenceVendors, "vendor"),
+        PORTSCAN(new String[] {}, evidencePortScanning, "portscan"),
+        COMMAND_EXEC(cmdExecPayloads, evidenceCmdExec, "command");
+
+        private String[] payloads;
+        private String[] responses;
+        private String resourceIdentifier;
+
+        XSLTInjectionType(String[] payloads, String[] responses, String resourceIdentifier) {
+            this.payloads = payloads;
+            this.responses = responses;
+            this.resourceIdentifier = resourceIdentifier;
+        }
+    }
+
+    private static final Logger LOG = Logger.getLogger(XSLTInjection.class);
+
+    // used to check against the attack strength
+    private int requestsSent = 0;
+    // map setting the limits associated with each attack strength
+    private static Map<AttackStrength, Integer> strengthToRequestCountMap = new HashMap<>();
+
+    static {
+        strengthToRequestCountMap.put(AttackStrength.LOW, 6);
+        strengthToRequestCountMap.put(AttackStrength.MEDIUM, 12);
+        strengthToRequestCountMap.put(AttackStrength.HIGH, 24);
+    }
+
+    @Override
+    public void scan(HttpMessage msg, String param, String value) {
+        // sets the payloads that recquire an instance to be determined
+        XSLTInjectionType.PORTSCAN.payloads = new String[] {getXslForPortScan()};
```

Comment:
This might lead to threading issues (e.g. one scanner sees the payload of other). The type could be changed to allow to return the payloads based on the message, e.g.: `List<String> getPayloads(HttpMessage)`, then `PORTSCAN` would call the existing method while the others would return the hardcoded lists.